app/components/add-instruction.js

```diff
@@ -151,6 +153,12 @@ export default class AddInstructionComponent extends Component {
       yield mapping.save();
     }
     yield this.template.save();
+
+    for (let i = 0; i < this.mappingsToBeDeleted.length; i++) {
```

Comment:
as Sam mentioned in claire's PR using the following is both shorter and (imo) more readable.
```
yield Promise.all(this.mappingsToBeDeleted.map((mapping) => mapping.destroyRecord())
```

Reply:
i remember you said there was a problem with promise.all and mu cl resources
so I was trying to avoid it
or was it mu auth

Reply:
...nvm